jobbergate-api/jobbergate_api/tests/apps/job_submissions/test_properties_parser.py

```diff
@@ -0,0 +1,657 @@
+import contextlib
+import inspect
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import MutableMapping
+from unittest import mock
+from pydantic import ValidationError
+
+import pytest
+from bidict import bidict
+
+from jobbergate_api.apps.job_scripts.job_script_files import JobScriptFiles
+from jobbergate_api.apps.job_submissions.properties_parser import (
+    _IDENTIFICATION_FLAG,
+    _INLINE_COMMENT_MARK,
+    ArgumentParserCustomExit,
+    _clean_jobscript,
+    _clean_line,
+    _flagged_line,
+    _split_line,
+    build_mapping_sbatch_to_slurm,
+    build_parser,
+    convert_sbatch_to_slurm_api,
+    get_job_parameters,
+    get_job_properties_from_job_script,
+    jobscript_to_dict,
+    sbatch_to_slurm,
+)
+from jobbergate_api.apps.job_submissions.schemas import JobProperties
+
+
+def test_identification_flag():
+    """
+    Check if the value of the identification flag is the same codded in the tests.
+    """
+    assert _IDENTIFICATION_FLAG == "#SBATCH"
+
+
+def test_inline_comment_mark():
+    """
+    Check if the inline comment mark is the same codded in the tests.
+    """
+    assert _INLINE_COMMENT_MARK == "#"
+
+
+@pytest.mark.parametrize(
+    "line, desired_value",
+    (
+        ("#SBATCH -abc", True),
+        ("##SBATCH -abc", False),
+        ("run python application", False),
+        ("# A comment", False),
+    ),
+)
+def test_flagged_line(line, desired_value):
+    """
+    Check if the flagged lines are identified properly.
+    """
+    actual_value = _flagged_line(line)
+    assert actual_value == desired_value
+
+
+@pytest.mark.parametrize(
+    "line, desired_value",
+    (
+        ("#SBATCH", ""),
+        ("#SBATCH#SBATCH", ""),
+        ("#SBATCH -abc # A comment", "-abc"),
+        ("#SBATCH --abc=0 # A comment", "--abc=0"),
+        ("#SBATCH --abc 0 # A comment", "--abc 0"),
+        ("#SBATCH -a 0 # A comment", "-a 0"),
+        ("#SBATCH -a=0 # A comment", "-a=0"),
+        ("#SBATCH -a = 0 # A comment", "-a = 0"),
+        ("#SBATCH -a=C&Aaccount # A comment", "-a=C&Aaccount"),
+    ),
+)
+def test_clean_line(line, desired_value):
+    """
+    Check if the provided lines are cleaned properly, i.e.,
+    identification flag, inline comments, and comment mark are all removed.
+    """
+    actual_value = _clean_line(line)
+    assert actual_value == desired_value
+
+
+@pytest.mark.parametrize(
+    "line, desired_value",
+    (
+        ("--help", ["--help"]),
+        ("--abc=0", ["--abc", "0"]),
+        ("--abc = 0", ["--abc", "0"]),
```

Comment:
this one shouldn't be even acceptable since slurm rejects it. See my screenshot:

<img width="456" alt="image" src="https://user-images.githubusercontent.com/45343359/205342482-6e049e07-909c-40d5-823b-a40a6d27bc0c.png">

Tested on Ubuntu 20.04. Deployed by the slurm charms

Reply:
fixed in [cfb1378](https://github.com/omnivector-solutions/jobbergate/pull/193/commits/cfb13783121304f9574f47be0ae4d859cb23aa1b)